### ISSUE TITLE
Rejected request on boot falls into infinite loop on SSR

### ISSUE THREAD

**Issue by Grawl** [quasarframework/quasar#13127]
### What happened?

I have to request some information at application boot phase, before something is rendered: I want this information in my templates. When server is unavailable, I mock this information. So I do not care if request fulfilled or rejected.

So, I do something like this in my `boot/axios.ts`:

```ts
import mockResponse from 'src/mock-response.ts'
const api = axios.create({ baseURL: 'http://localhost:8080/api' })
export default boot(async () => {
  const store = useStore()
  await api.get('/bad-request')
    .then(response => {
      store.setData(response)
    })
    .catch(error => {
      store.setData(mockResponse)
      console.error({ error })
    })
})
```

Note: `baseURL` is the same host as my app because I use some cookies. Server is available using Webpack/Nginx proxy: `http://localhost:3000` (backend URL) is proxied from `http://localhost:8080/api` (frontend URL)

But when server is unavailable, I have infinite loading tab instead of working application after request is rejected.

Debugging this, I figured out that if I set `timeout: 1000` for my Axios instance, I have infinite errors in SSR console slowing rendering. After dozens of errors in console, I have working frontend in my browser, but SSR console still showing me these errors again and again, repeating requests to unavailable server.

### What did you expect to happen?

SSR console should show me one error for one request I do, not infinite errors. There should be one request, not infinite ones.

### Reproduction URL

https://github.com/Grawl/quasar-ssr-boot-bad-request

### How to reproduce?

1. Clone repository
2. `yarn install`
3. `yarn build`
4. `yarn start`
5. Look at console where you run previous command, you should see `Server listening at port 8080`, I call it "SSR console"
6. Open `http://localhost:8080/` in your browser
7. Look back at SSR console, you will see infinite errors

### Flavour

Quasar CLI with Webpack (@quasar/cli | @quasar/app-webpack)

### Areas

SSR Mode

### Quasar info output

```Shell
Operating System - Darwin(21.1.0) - darwin/x64
NodeJs - 16.13.1

Global packages
  NPM - 8.1.2
  yarn - 1.22.17
  @quasar/cli - undefined
  @quasar/icongenie - Not installed
  cordova - Not installed

Important local packages
  quasar - 2.6.5 -- Build high-performance VueJS user interfaces (SPA, PWA, SSR, Mobile and Desktop) in record time
  @quasar/app-webpack - 3.4.7 -- Quasar Framework App CLI with Webpack
  @quasar/extras - 1.13.5 -- Quasar Framework fonts, icons and animations
  eslint-plugin-quasar - Not installed
  vue - 3.2.31 -- The progressive JavaScript framework for building modern web UI.
  vue-router - 4.0.14
  pinia - Not installed
  vuex - Not installed
  electron - Not installed
  electron-packager - Not installed
  electron-builder - Not installed
  @babel/core - 7.17.9 -- Babel compiler core.
  webpack - 5.72.0 -- Packs CommonJs/AMD modules for the browser. Allows to split your codebase into multiple bundles, which can be loaded on demand. Support loaders to preprocess files, i.e. json, jsx, es7, css, less, ... and your custom stuff.
  webpack-dev-server - 4.7.4 -- Serves a webpack app. Updates the browser on changes.
  workbox-webpack-plugin - Not installed
  register-service-worker - 1.7.2 -- Script for registering service worker, with hooks
  typescript - 4.5.5 -- TypeScript is a language for application scale JavaScript development
  @capacitor/core - Not installed
  @capacitor/cli - Not installed
  @capacitor/android - Not installed
  @capacitor/ios - Not installed

Quasar App Extensions
  *None installed*

Networking
  Host - iMac-Daniil.local
  en1 - 10.23.18.82
```

### Relevant log output

```Shell
{
  error: Error: timeout of 1000ms exceeded
      at createError (/Users/daniilpronin/Sites/quasar-project/node_modules/axios/lib/core/createError.js:16:15)
      at RedirectableRequest.handleRequestTimeout (/Users/daniilpronin/Sites/quasar-project/node_modules/axios/lib/adapters/http.js:303:16)
      at RedirectableRequest.emit (node:events:390:28)
      at Timeout.<anonymous> (/Users/daniilpronin/Sites/quasar-project/node_modules/follow-redirects/index.js:164:12)
      at listOnTimeout (node:internal/timers:557:17)
      at processTimers (node:internal/timers:500:7) {
    config: {
      url: '/bad-request',
      method: 'get',
      headers: [Object],
      baseURL: 'http://localhost:8080/api',
      transformRequest: [Array],
      transformResponse: [Array],
      timeout: 1000,
      adapter: [Function: httpAdapter],
      xsrfCookieName: 'XSRF-TOKEN',
      xsrfHeaderName: 'X-XSRF-TOKEN',
      maxContentLength: -1,
      maxBodyLength: -1,
      validateStatus: [Function: validateStatus],
      transitional: [Object],
      data: undefined
    },
    code: 'ECONNABORTED',
    request: Writable {
      _writableState: [WritableState],
      _events: [Object: null prototype],
      _eventsCount: 3,
      _maxListeners: undefined,
      _options: [Object],
      _ended: true,
      _ending: true,
      _redirectCount: 0,
      _redirects: [],
      _requestBodyLength: 0,
      _requestBodyBuffers: [],
      _onNativeResponse: [Function (anonymous)],
      _currentRequest: [ClientRequest],
      _currentUrl: 'http://localhost:8080/api/bad-request',
      _timeout: null,
      [Symbol(kCapture)]: false
    },
    response: undefined,
    isAxiosError: true,
    toJSON: [Function: toJSON]
  }
}
```

**Comment by rstoenescu** [comment]
Thank you for the detailed report and the repro. Traced it, and this is not a Quasar bug: the SSR server is requesting itself, and every one of those requests triggers another full page render.

What happens in the repro:

1. The production SSR server listens on port 8080, and the axios instance in `src/boot/axios.ts` uses `http://localhost:8080/api` as its `baseURL`. On the server, that is the SSR server's own address.
2. There is no handler for `/api` in `src-ssr/middlewares`. The Webpack `devServer.proxy` only exists while running `quasar dev`; it is not part of the production server started by `yarn build` + `yarn start`.
3. So `GET /api/bad-request` falls through to the catch-all in `src-ssr/middlewares/render.js`, which renders the app, which runs the boot file, which sends `GET /api/bad-request` to the same server again. That render awaits its own child request, and so on.
4. With `timeout: 1000`, each level gives up after one second and continues, but by then its child render has already spawned a grandchild. A new request fires roughly every second, forever: that is the endless error stream in the SSR console. Without the timeout, each level waits on its child indefinitely, which is the "infinite loading tab" variant.
5. After the browser finally receives a page, the client-side boot sends the same request to port 8080, which starts a fresh server-side chain. That is why the frontend works while the console keeps erroring.

The fix is on the app side, either of:

- Mount an `/api` handler (or a proxy middleware to your backend) in `src-ssr/middlewares` and list it before `render` in `quasar.config.js > ssr.middlewares`, so API requests never reach the page-render catch-all.
- Point the server-side axios instance directly at the backend (for example `baseURL: process.env.SERVER ? 'http://localhost:3000' : '/api'`) instead of going through the SSR server's own origin.

As a general rule, any code that runs during server-side rendering must never request the SSR server itself for a URL that the render middleware handles, otherwise the render recurses.

Closing as not planned.
